### PR TITLE
single post component

### PR DESCRIPTION
**Pull request by ZohaibManzoor00**
Created a reusable post component and updated ui to use dark mode 

Light mode 
Screenshot (250)

Dark mode
Screenshot (251)

### CODE REVIEW

File: src/shared/components/post/Post.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  readonly id: number;
+  readonly userId: number;
+  readonly profilePic?: string;
+  readonly firstName: string;
+  readonly lastName: string;
+  readonly timestamp?: Date,
+  readonly content: string;
+  readonly media?: string;
+};
+
```
Comment (DLiSWE) [inline src/shared/components/post/Post.tsx:17]: Maybe leave this model data in another folder

**Comment by Matt-Tan15** [comment]
I think instead of being able to add and remove friends on the top right we should put a dropdown with 3 dots for like editing and delete posts since it's too easy to add and accidently remove a friend if we keep it as it is rn. We could add a hover over the user's name and show this instead for adding and removing friends 
image

File: src/shared/components/post/Post.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        padding: 0,
+        boxSizing: 'border-box'
+      }}
+    >
+      <Box
+        component="div"
+        sx={{
+          mr: "20px",
+          ml: "20px",
+        }}
```
Comment (DLiSWE) [inline src/shared/components/post/Post.tsx:38]: mx: 20px

File: src/shared/components/post/PostTop.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+                  backgroundColor: navButtons,
+                  width: "14ch",
+                  color,
+                },
+              }}
+            >
+              {options.map((option) => (
+                <MenuItem
+                  key={option.title}
+                  selected={option.title === "Pyxis"}
```
Comment (DLiSWE) [inline src/shared/components/post/PostTop.tsx:91]: What is pyxis.

Comment (ZohaibManzoor00) [reply src/shared/components/post/PostTop.tsx]: This is actually unnecessary, good catch!

**Review by Matt-Tan15 — APPROVED** [review]
nice work!

**Review by DLiSWE — APPROVED** [review]

**Pull request merged**